src/ceph-daemon/ceph-daemon

```diff
@@ -411,19 +457,26 @@ def get_legacy_daemon_fsid(cluster, daemon_type, daemon_id, legacy_dir=None):
 
 def get_daemon_args(fsid, daemon_type, daemon_id):
     # type: (str, str, Union[int, str]) -> List[str]
-    r = [
-        '--default-log-to-file=false',
-        '--default-log-to-stderr=true',
+    r = []
+
+    if daemon_type in Ceph.daemons:
+        r += [
+            '--default-log-to-file=false',
+            '--default-log-to-stderr=true',
+            '--setuser', 'ceph',
+            '--setgroup', 'ceph'
         ]
-    r += ['--setuser', 'ceph']
-    r += ['--setgroup', 'ceph']
+
+    elif daemon_type in Monitoring.components:
+        metadata = Monitoring.components[daemon_type]['image']
+        r += metadata['args']
     return r
 
 def create_daemon_dirs(fsid, daemon_type, daemon_id, uid, gid,
                        config=None, keyring=None):
     # type: (str, str, Union[int, str], int, int, str, str) ->  None
-    data_dir = make_data_dir(fsid, daemon_type, daemon_id)
-    make_log_dir(fsid)
+    data_dir = make_data_dir(fsid, daemon_type, daemon_id, uid, gid)
+    make_log_dir(fsid, uid, gid)
```

Comment:
this hunk looks like a bug independent of this pr?

Reply:
I had to fix it for this PR - since the uid/gid is set by the default extract_uid_gid which expects the image to be ceph. For monitoring it's not - so it had to be fixed.

Reply:
ha! looks as if we are attempting to solve the same thing: https://github.com/ceph/ceph/pull/32137/commits/5111a1145946a7b6005a68ec7e9bf19b88f60524#diff-0d7e098eacbe52f3a7f86d925dfeff1fR450

@pcuzner I'll drop my change in favor of yours, but instead positional args can we specify the keywords?
```suggestion
    data_dir = make_data_dir(fsid, daemon_type, daemon_id, uid=uid, gid=gid)
    make_log_dir(fsid, uid=uid, gid=gid)
```

Reply:
err, actually sorry, guess PR #32137 was just merged. But :+1: for adding uid/gid to `make_log_dir`